File: frontend/configs/Personas.ts
export const Personas = {
  endpoint: 'personas',
  columnas: [
    { key: 'dni', label: 'DNI' },
    { key: 'nombres', label: 'Nombres' },
    { key: 'apellido_paterno', label: 'Apellido Paterno' },
  ],
  camposFormulario: [
    { key: 'tipo_documento', label: 'Tipo de Documento', type: 'text' },
    { key: 'dni', label: 'DNI', type: 'text' },
    { key: 'nombres', label: 'Nombres', type: 'text' },
    { key: 'apellido_paterno', label: 'Apellido Paterno', type: 'text' },
    { key: 'apellido_materno', label: 'Apellido Materno', type: 'text' },
    { key: 'fecha_nacimiento', label: 'Fecha de Nacimiento', type: 'date' },
    {
        key: 'genero',
        label: 'Género',
        type: 'select',
        options: [
        { label: 'Otro', value: 'O' },
        { label: 'Masculino', value: 'M' },
        { label: 'Femenino', value: 'F' }
        ]
    },
    { key: 'direccion_cifrada', label: 'Dirección (Texto plano)', type: 'text' },
    { key: 'telefono_cifrado', label: 'Teléfono (Texto plano)', type: 'text' },
    { key: 'correo_cifrado', label: 'Correo (Texto plano)', type: 'email' },
  ]
}

  /*relaciones: [
    { key: 'rol_id', recurso: 'roles', label: 'nombre' },
    { key: 'permiso_id', recurso: 'permisos', label: 'nombre' },
    { key: 'creado_por', recurso: 'personas', label: 'username' }
  ],*/

File: backend/schemas/personas.py
from pydantic import BaseModel, constr
from typing import Optional
from datetime import date, datetime

class PersonaBase(BaseModel):
    tipo_documento: constr(max_length=20)
    dni: constr(max_length=15)
    nombres: constr(max_length=100)
    apellido_paterno: constr(max_length=100)
    apellido_materno: Optional[constr(max_length=100)] = None
    fecha_nacimiento: date
    genero: Optional[str] = 'O'

class PersonaCreate(PersonaBase):
    direccion_cifrada: Optional[bytes] = None
    telefono_cifrado: Optional[bytes] = None
    correo_cifrado: Optional[bytes] = None

class PersonaRead(PersonaBase):
    id: int
    direccion_cifrada: Optional[bytes] = None
    telefono_cifrado: Optional[bytes] = None
    correo_cifrado: Optional[bytes] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True
